src/domains/chart/components/lib-charts/alarmBadge.js

```diff
@@ -0,0 +1,54 @@
+import React, { forwardRef } from "react"
+import styled from "styled-components"
+
+export const getBackgroundColor = (status) => ({
+  WARNING: "#FFF8E1",
+  CRITICAL: "#FFEBEF",
+  CLEAR: "#E5F5E8",
+})[status] || null
+
+export const getBorderColor = (status) => ({
+  WARNING: "#FFC300",
+  CRITICAL: "#F59B9B",
+  CLEAR: "#68C47D",
+})[status] || null
```

Comment:
This pattern is not so optimal since a new object is created in every execution. 
You could move out the hash map definition so it will be executed once on the initialization. Something like:

const borderColorMap = {warning: '...'}

getBorderColor = borderColorMap[status] || borderColorMap.clear

Reply:
Done, although maybe we should talk about such cases, I would like to learn about your reasoning, i see it as micro- (or even nano-) optimisation, and was focusing only on code readability